pkg/storage/memory/memory.go

```diff
@@ -21,57 +22,48 @@ import (
 var tracer = otel.Tracer("openfga/pkg/storage/memory")
 
 type staticIterator struct {
-	tuples            []*openfgav1.Tuple
+	records           []*storage.TupleRecord
 	continuationToken []byte
 	mu                sync.Mutex
 }
 
-func match(key *openfgav1.TupleKey, target *openfgav1.TupleKey) bool {
-	if key.Object != "" {
-		td, objectid := tupleUtils.SplitObject(key.Object)
-		if objectid == "" {
-			if td != tupleUtils.GetType(target.Object) {
-				return false
-			}
-		} else {
-			if key.Object != target.Object {
-				return false
-			}
-		}
-	}
-	if key.Relation != "" && key.Relation != target.Relation {
-		return false
-	}
-	if key.User != "" && key.User != target.User {
-		return false
-	}
-	return true
-}
-
 func (s *staticIterator) Next() (*openfgav1.Tuple, error) {
```

Comment:
It'd be nice if we could get rid of this duplicate `staticIterator` in favor of a usage pattern that uses the other generic static iterator in `pkg/storage/iterators.go`.

Do you see an elegant way to do so? If so, let's go that direction.

Reply:
I can do that in a separate PR